### PR TITLE
clang-tidy: Apply modernize-use-override.

### PR DESCRIPTION
**Pull request by aarlt**

### CODE REVIEW

File: libsolidity/analysis/GlobalContext.cpp [hunk 1]
```diff
@@ -123,7 +123,7 @@ vector<Declaration const*> GlobalContext::declarations() const
 {
 	vector<Declaration const*> declarations;
 	declarations.reserve(m_magicVariables.size());
-	for (ASTPointer<Declaration const> const& variable: m_magicVariables)
```
Comment (aarlt) [inline libsolidity/analysis/GlobalContext.cpp]: Hmm.. strange that this was also changed by `modernize-use-override`. probably we need to remove this change.

Comment (ekpyron) [reply libsolidity/analysis/GlobalContext.cpp]: Yeah... and it's not a good change... this means that we'll really have to carefully review all of these :-).

File: libyul/optimiser/NameCollector.cpp [hunk 2]
```diff
@@ -36,9 +36,9 @@ void NameCollector::operator()(VariableDeclaration const& _varDecl)
 void NameCollector::operator ()(FunctionDefinition const& _funDef)
 {
 	m_names.emplace(_funDef.name);
-	for (auto const arg: _funDef.parameters)
+	for (auto const &arg: _funDef.parameters)
```
Comment (aarlt) [inline libyul/optimiser/NameCollector.cpp]: I will leave that change.

File: libyul/optimiser/TypeInfo.cpp [hunk 3]
```diff
@@ -50,12 +50,12 @@ class TypeInfo::TypeCollector: public ASTWalker
 		ASTWalker::operator()(_funDef);
 
 		auto& funType = functionTypes[_funDef.name];
-		for (auto const arg: _funDef.parameters)
+		for (auto const &arg: _funDef.parameters)
```
Comment (aarlt) [inline libyul/optimiser/TypeInfo.cpp]: I will also leave that change.

**Review by chriseth — APPROVED** [review]

**Pull request merged**